app/helpers/application_helper.rb

```diff
@@ -0,0 +1,17 @@
+module ApplicationHelper
+  def signin(post)
+    if user_signed_in?
+      p post.user.username
+    else
+      render plain: 'Anon'
+    end
+  end
+
+  def login
+    if user_signed_in?
+      render 'signedout'
+    else
+      render 'signedin'
```

Comment:
Just as highlighted by the previous Tse's, your logic is returning a string instead of a boolean. View doesn't know where to find this, hence the missing template error.